Api script schema

hshiue opened the pull request:
API script that generates token, fetch and write content into three types of files. .ini filenames, token script, namespace and urls are all hardcoded.

nkrabben commented on bin/get_schemas.py:
```diff
@@ -0,0 +1,106 @@
+        '-i',
+        type=str,
+        required=True,
+        choices=['test', 'prod'],
```
I added the choices arg here. It can be very handy. Automatically tests if the input is in the list.

nkrabben commented on bin/get_schemas.py:
```diff
@@ -0,0 +1,106 @@
+    return parser.parse_args()
+
+def generate_access_token(config_input: str):
+    config = configparser.ConfigParser()
```
I don't understand if the `config. ...` lines are necessary. Seems like you read a file and then don't do anything with it

nkrabben reviewed (approved):
I made a commit to argparse and had questions about the token function.

We need to start planning how all of these scripts work together, especially if there are components that we're reusing.

hshiue commented:
This has been updated! Nick, you are correct. The whole
function, generate_access_token, is not needed. I removed it and the script
just generates the token in main.

On Fri, Jun 30, 2023 at 2:23 PM Nick Krabbenhoeft ***@***.***>
wrote:

> ***@***.**** approved this pull request.
>
> I made a commit to argparse and had questions about the token function.
>
> We need to start planning how all of these scripts work together,
> especially if there are components that we're reusing.
> ------------------------------
>
> In bin/get_schemas.py
> <https://github.com/NYPL/prsv-tools/pull/9#discussion_r1248157386>:
>
> > +import configparser
> +import xml.etree.ElementTree as ET
> +from pathlib import Path
> +
> +# prsvtoken.py needs to be in the same directory for this to work
> +import prsvtoken
> +
> +def parse_args():
> +    parser = argparse.ArgumentParser()
> +
> +    parser.add_argument(
> +        '--instance',
> +        '-i',
> +        type=str,
> +        required=True,
> +        choices=['test', 'prod'],
>
> I added the choices arg here. It can be very handy. Automatically tests if
> the input is in the list.
> ------------------------------
>
> In bin/get_schemas.py
> <https://github.com/NYPL/prsv-tools/pull/9#discussion_r1248158206>:
>
> > +        choices=['test', 'prod'],
> +        help='Which Preservica instance to access'
> +    )
> +
> +    parser.add_argument(
> +        '--destination_folder_path',
> +        '-dest',
> +        type=str,
> +        required=False,
> +        help='''Optional. Provide an absolute folder path to save the files
> +        in the specified folder'''
> +    )
> +    return parser.parse_args()
> +
> +def generate_access_token(config_input: str):
> +    config = configparser.ConfigParser()
>
> I don't understand if the config. ... lines are necessary. Seems like you
> read a file and then don't do anything with it
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NYPL/prsv-tools/pull/9#pullrequestreview-1507687855>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYEXVZ57MHDYWLC5C7R6N7TXN4KTXANCNFSM6AAAAAAZ2AD4MQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

*Hilary Shiue (she/her/hers)*
Digital Repository Coordinator
Digital Research Division

Library Services Center

*The New York Public Library *31-11 Thomson Avenue, Long Island City, NY
11101

917.229.9580 | x39580
